x-pack/plugin/mapper-unsigned-long/src/main/java/org/elasticsearch/xpack/unsignedlong/UnsignedLongFieldMapper.java

```diff
@@ -626,7 +626,15 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
             numericValue = null;
         } else {
             try {
-                if (parser.currentToken() == XContentParser.Token.VALUE_NUMBER) {
+                if (parser.currentToken().isValue() == false) {
+                    if (ignoreMalformed.value()) {
+                        context.addIgnoredField(mappedFieldType.name());
+                        if (isSourceSynthetic) {
+                            context.doc().add(IgnoreMalformedStoredValues.storedField(fullPath(), context.parser()));
+                        }
```

Comment:
Are we missing a return here? Right now it will still throw an exception with ignore malformed enabled.

Reply:
I took a closer look and there is a problem that when the source is not synthetic we'll not advance the parser to the end of the object and fail later with parsing exception (`found extra data after parsing: END_OBJECT`). I think this is actually a reason that we have `parser.currentToken().isValue()` check when handling `ignore_malformed`. 

It looks like my expectations in #109705 are incorrect and this works as designed - "value fields" like numbers don't take objects as inputs even with `ignore_malformed`. If we wanted to change that we probably need a wider group decision. What do you think?

Reply:
I do not expect non-object fields to accept object-like values...but then we need to agree on how to handle them when it comes to `ignore_malformed`.

Reply:
On the other end, anyway, I think the purpose of `ignore_malformed` is to avoid documents being rejected because of a a malformed field value. So I think the right behaviour would be to just parse the object until the end so to avoid the assertion failure later but then add the field to ignored fields and store its value.